Content.Server/Station/Systems/StationSystem.cs

```diff
@@ -320,11 +304,32 @@ public EntityUid InitializeNewStation(StationConfig stationConfig, IEnumerable<E
         var data = Comp<StationDataComponent>(station);
         name ??= MetaData(station).EntityName;
 
+
         foreach (var grid in gridIds ?? Array.Empty<EntityUid>())
         {
             AddGridToStation(station, grid, null, data, name);
         }
 
+        if (TryComp<StationRandomComponent>(station, out var random))
+        {
+            var rotation = _random.NextAngle();
+            var offset = new Vector2(
+                _random.NextFloat(-random.MaxStationOffset, random.MaxStationOffset),
+                _random.NextFloat(-random.MaxStationOffset, random.MaxStationOffset));
+
+            foreach (var grid in gridIds ?? Array.Empty<EntityUid>())
+            {
+                if (random.EnableStationRotation)
+                    _transform.SetWorldRotation(grid, rotation);
+
+                if (random.EnableStationOffset)
+                {
+                    var pos = _transform.GetWorldPosition(grid);
+                    _transform.SetWorldPosition(grid, pos + offset);
+                }
+            }
+        }
+
```

Comment:
1. You should be setting the transform in 1 call as this is going to send out double the moveevents it needs to.
2. You should be re-using the transformcomponent.
3. The array coalescing looks like it was copy-pasted so just make a local variable for it.